source/val/validate_constants.cpp

```diff
@@ -33,12 +33,18 @@ spv_result_t ValidateConstantBool(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+bool isCompositeType(const Instruction* inst) {
+  return spvOpcodeIsComposite(inst->opcode()) ||
+         (inst->opcode() == spv::Op::OpTypeTensorARM &&
```

Comment:
Right.  It has a Shape operand.  As per the extension spec, a tensor type is composite if and only if it has a shape operand.

Reply:
I'll add a boolean variable or leave a comment to make this easier to understand.